src/dogecoin-fees.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021 The Dogecoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <boost/random/uniform_int.hpp>
+
+
+#include "policy/policy.h"
+#include "arith_uint256.h"
+#include "dogecoin.h"
+#include "txmempool.h"
+#include "util.h"
+#include "validation.h"
+#include "dogecoin-fees.h"
+#include "amount.h"
+#include <cmath>
+
+//mlumin: concept:  min relay fee 0.0001 doge; 
+//mlumin: concept:  min fee configurable and defaulted at 0.01 DOGE later
+
+
+//mlumin 5/2021: walletfees, all attached to GetDogecoinWalletFeeRate which is just the newly exposed ::minWalletTxFee
+CAmount GetDogecoinWalletFee(size_t nBytes_)
+{
+    //mlumin: super simple fee calc for dogecoin
+    //CAmount nFee = ((int)ceil(nBytes_/1000.0f));
+    CAmount nFee=GetDogecoinWalletFeeRate().GetFee(nBytes_);
+}
+
+CAmount GetDogecoinWalletFeeWithDust(const CTransaction& tx, unsigned int nBytes)
+{
+    CFeeRate walletFeeRate=GetDogecoinWalletFeeRate();
+    //mlumin: get the staged or created tx from the send ui - not sure we need to lock mempool in this case.
+    {
+        LOCK(mempool.cs);//mlumin: do we need to lock the mempool when preparing to transmit a wallet tx? 
+        uint256 hash = tx.GetHash();
+        double dPriorityDelta = 0;
+        CAmount nFeeDelta = 0;
+        mempool.ApplyDeltas(hash, dPriorityDelta, nFeeDelta);
+        if (dPriorityDelta > 0 || nFeeDelta > 0)
+            return 0;
+    }
```

Comment:
How does this impact fee calculation when creating a transaction? (now and in the future)

(also: duplicate code with L62-72)

Reply:
This is the free transaction enablement logic isn't it?